includes/sanitizers/class-amp-style-sanitizer.php

```diff
@@ -1969,7 +1969,7 @@ private function ampify_ruleset_selectors( $ruleset ) {
 				foreach ( $edited_selectors as &$edited_selector ) { // Note: The $edited_selectors array contains only item in the normal case.
 					$original_selector = $edited_selector;
 					$amp_selector      = array_shift( $amp_selectors );
-					$amp_tag_pattern   = '/(?<=^|[^a-z0-9_-])' . preg_quote( $amp_selector, '/' ) . '(?=$|[^a-z0-9_-])/i';
+					$amp_tag_pattern   = '/(?<=^|\s|>)' . preg_quote( $amp_selector, '/' ) . '(?=$|[^a-z0-9_-])/i';
```

Comment:
Also need to account for other possible characters which can precede an element name. I'm thinking in particular:

* Adjacent sibling combinator (`+`)
* General sibling combinator (`~`)

Once PHP-CSS-Parser is updated to actually parse the components of a selector, we can be smarter about the replacements instead of using regular expressions.